src/runtime.rs

```diff
@@ -133,6 +135,37 @@ impl Runtime {
         unsafe { qjs::JS_RunGC(guard.rt) }
         mem::drop(guard);
     }
+
+    /// Test for pending jobs
+    ///
+    /// Returns true when at least one job is pending.
+    pub fn is_job_pending(&self) -> bool {
+        let guard = self.inner.lock();
+        let res = unsafe { qjs::JS_IsJobPending(guard.rt) };
+        mem::drop(guard);
+        res != 0
+    }
+
+    /// Execute first pending job
+    ///
+    /// Returns context for executed job or none when queue is empty or error when exception thrown under execution.
+    pub fn execute_pending_job(&self) -> Result<Option<Ctx<'_>>, Error> {
```

Comment:
`Ctx` should not be available outside a callback handed to `Context::with` otherwise the runtime can be accessed outside the mutex gaurd, which is unsafe

Reply:
This functions should probably be a member of `Ctx`.

Reply:
By design QuickJS has single queue per runtime, so this function cannot be member of `Ctx`.

Reply:
My next pr #6 supersede this, I already changed returning value.